File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        result = [1]
        prod = 1
        for i in range(1, len(nums)):
            prod *= nums[i-1]
            result.append(prod)
        prod = 1
        for i in range(len(nums)-1, -1, -1):
            if i == len(nums)-1:continue
            prod *= nums[i+1]
            result[i] = result[i]*prod
        return result

File: 238-product-of-array-except-self/238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */       
var productExceptSelf = function(nums) {
    let result = [1]
    let prod = 1
    for(i=1; i<nums.length; i++){
        prod *= nums[i-1]
        result.push(prod)
    }
    prod = 1
    for(i=nums.length-1; i>-1; i--){
        if (i==nums.length-1){continue}
        prod = prod*nums[i+1]
        result[i] = result[i]*prod
    }
    return result
};